src/baseCanvas/Canvas.tsx

```diff
@@ -20,18 +26,9 @@ export interface Props {
   onMouseMove?: (x: number, y: number) => void;
   onMouseUp?: (x: number, y: number) => void;
   onContextMenu?: (x: number, y: number) => void;
-  canvasPositionAndSize: {
-    top: number;
-    left: number;
-    width: number;
-    height: number;
-  };
-  setCanvasPositionAndSize?: (canvasPositionAndSize: {
-    top?: number;
-    left?: number;
-    width?: number;
-    height?: number;
-  }) => void;
+  canvasPositionAndSize: PositionAndSize;
```

Comment:
```suggestion
  canvasPositionAndSize: Required<PositionAndSize>;
```

(Nice to enforce it if none of the values are actually optional here)

Reply:
They are all optional in the interface I think?

Reply:
```
export interface PositionAndSize {
  top?: number;
  left?: number;
  width?: number;
  height?: number;
}
```

Reply:
Yea, so this effective makes it

```typescript
interface PositionAndSize {
  top: number;
  left: number;
  width: number;
  height: number;
}
``` when used here

Reply:
Ah, I see... got ya...